client/go/outline/config/module.go

```diff
@@ -98,8 +98,15 @@ func NewDefaultTransportProvider(tcpDialer transport.StreamDialer, udpDialer tra
 	packetEndpoints.RegisterSubParser("websocket", NewWebsocketPacketEndpointSubParser(streamEndpoints.Parse))
 
 	// Stream dialers.
+	streamDialers.RegisterSubParser("direct", func(ctx context.Context, input map[string]any) (*Dialer[transport.StreamConn], error) {
+		return &Dialer[transport.StreamConn]{
```

Comment:
Line 42 above, where we define the streamDialers should be using the same dialer. Let's share the object.